src/backend/distributed/executor/executor.c

```diff
@@ -2390,20 +2466,24 @@ GetTaskExecutionState(ShardCommandExecution *shardCommandExecution)
 
 	if (failedPlacementCount == placementCount)
 	{
-		return TASK_EXECUTION_FAILED;
+		currentTaskExecutionState = TASK_EXECUTION_FAILED;
 	}
 	else if (executionOrder == EXECUTION_ORDER_ANY && donePlacementCount > 0)
 	{
-		return TASK_EXECUTION_FINISHED;
+		currentTaskExecutionState = TASK_EXECUTION_FINISHED;
 	}
 	else if (donePlacementCount + failedPlacementCount == placementCount)
 	{
-		return TASK_EXECUTION_FINISHED;
+		currentTaskExecutionState = TASK_EXECUTION_FINISHED;
 	}
 	else
 	{
-		return TASK_EXECUTION_NOT_FINISHED;
+		currentTaskExecutionState = TASK_EXECUTION_NOT_FINISHED;
 	}
+
+	shardCommandExecution->executionState = currentTaskExecutionState;
```

Comment:
Should we rename this function to something other than `Get`?

Reply:
Maybe only `TaskExecutionState()` ?

Reply:
It kind of seems like we're actually moving the task execution state machine forward here. Maybe `TaskExecutionStateMachine` to be in line with the other functions.